targets/cloudscript-typings/make.js

```diff
@@ -26,3 +28,122 @@ exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
     writeFile(path.resolve(outputDir, "CloudScript.d.ts"), generatedApi);
 
 }
+
+/**
+  Handles selecting the correct datatype template to use for a datatype.
+  enum, interface
+*/
+function MakeDatatype(datatype, api, sourceDir) {
+  var stringLiteralTemplate = compileTemplate(sourceDir,"StringLiteral");
+  var interfaceTemplate = compileTemplate(sourceDir,"Interface");
+
+  var locals = {
+    name: datatype.name,
+  };
+
+  if(datatype.isenum) {
+    locals.enumvalues = datatype.enumvalues;
+    return  GetDescription(datatype.description, "")+ stringLiteralTemplate(locals);
+  }
+  else {
+    locals.properties = datatype.properties;
+    locals.sourceDir = sourceDir;
+    locals.api = api;
+    locals.GetProperty = GetProperty;
+    return  GetDescription(datatype.description, "") + interfaceTemplate(locals);
+  }
+
+}
+
+/** Handles generating a property field for inside an interface */
+function GetProperty(property, api, sourceDir) {
+
+  var type = property.jsontype.toLowerCase();
+
+  if(type === "object") {
+    type = "any";
+  }
+  if(property.isenum || property.isclass) {
+    type = property.actualtype;
+  }
+
+  var preColon = property.name + (property.optional?"?":"");
+  var postColon = type;
+
+  if(property.collection === "map") {
+    postColon = "{ [key:string] : "+ type +"}";
+  }
+  if(property.collection === "array") {
+    postColon += "[]";
+  }
+
+  return GetDescription(property.description, "   ") + "   "+ preColon + " : " + postColon + ",\n";
```

Comment:
You've hard coded your tab-spacing here.
I'd add a parameter that makes the spacing an input parameter of GetProperty, the same way as GetDescription.